backend/api/response.go

```diff
@@ -78,3 +78,19 @@ type userRequestsResponse struct {
 }
 
 // ========================================================================
+
+// ========================================================================
+// Type and function to return users active errand with its requests.
+type activeErrandResponse struct {
+	Errand   db.Errand                     `json:"errand"`
+	Requests []activeErrandRequestResponse `json:"requests"`
+}
+
+type activeErrandRequestResponse struct {
+	Request db.Request   `json:"request"`
+	Items   []db.Item    `json:"items"`
+	User    userResponse `json:"user"`
+	Store   *db.Store    `json:"store"`
```

Comment:
Do we need to make this a pointer? And will that mess up the actual response, like send a pointer to the store rather than the actual store data?

Reply:
a store is not required for an item so yeah it could be null. I checked by making store_id null when testing and it just sends back "store": null if there is no store which should be easy for frontend to handle.